File: examples/CSharp/DataFeed/DataFeedTest.cs
using System;
using System.Threading.Tasks;

using NBomber;
using NBomber.Contracts;
using NBomber.CSharp;

namespace CSharp.DataFeed
{
    public class User
    {
        public int Id { get; set; }
        public string Name { get; set; }
    }

    public class DataFeedTest
    {
        public static void Run()
        {
            var data = FeedData.FromSeq(new[] {1, 2, 3, 4, 5}).ShuffleData();
            //var data = FeedData.FromJson<User>("./DataFeed/users-feed-data.json");
            //var data = FeedData.FromCsv<User>("./DataFeed/users-feed-data.csv");

            var feed = Feed.CreateCircular("numbers", data);
            //var feed = Feed.CreateConstant("numbers", data);
            //var feed = Feed.CreateRandom("numbers", data);

            var step = Step.Create("step", feed, async context =>
            {
                await Task.Delay(TimeSpan.FromSeconds(1));

                context.Logger.Debug("Data from feed: {FeedItem}", context.FeedItem);
                return Response.Ok();
            });

            var scenario = ScenarioBuilder.CreateScenario("data_feed_scenario", step);

            NBomberRunner
                .RegisterScenarios(scenario)
                .Run();
        }
    }
}

File: examples/FSharp/DataFeed/DataFeedTest.fs
module FSharp.DataFeed.DataFeedTest

open System.Threading.Tasks

open FSharp.Control.Tasks.NonAffine

open NBomber
open NBomber.Contracts
open NBomber.FSharp

[<CLIMutable>]
type User = {
    Id: int
    Name: string
}

let run () =

    let data = [1; 2; 3; 4; 5] |> FeedData.fromSeq |> FeedData.shuffleData
    //let data = FeedData.fromJson<User>("./DataFeed/users-feed-data.json")
    //let data = FeedData.fromCsv<User>("./DataFeed/users-feed-data.csv")

    let feed = data |> Feed.createCircular "numbers"
    //let feed = data |> Feed.createConstant "numbers"
    //let feed = data |> Feed.createRandom "numbers"

    let step = Step.create("step", feed, fun context -> task {

        do! Task.Delay(seconds 1)

        context.Logger.Debug("Data from feed: {FeedItem}", context.FeedItem)
        return Response.Ok()
    })

    Scenario.create "data_feed_scenario" [step]
    |> NBomberRunner.registerScenario
    |> NBomberRunner.run
    |> ignore
